internal/tests/cluster/sequential/session_cleanup_test.go

```diff
@@ -366,8 +373,17 @@ func testWorkerSessionCleanupMulti(burdenCase timeoutBurdenType) func(t *testing
 		event.WriteSysEvent(ctx, op, "resuming connections to both controllers")
 		p1.Resume()
 		p2.Resume()
-		err = w1.Worker().WaitForNextSuccessfulStatusUpdate()
-		require.NoError(err)
+
+		wg.Add(2)
+		go func() {
+			defer wg.Done()
+			helper.ExpectWorkers(t, c1, w1)
+		}()
+		go func() {
+			defer wg.Done()
+			helper.ExpectWorkers(t, c2, w1)
+		}()
+		wg.Wait()
```

Comment:
Is this equivalent to the old code?

Reply:
I think the old code was flaky. I believe that we actually want to verify that both controller 1 & 2 are able to connect to the worker. The reason why I think this is the expected behavior is because both proxy 1 and 2 were resumed at the same time.